src/language-js/printer-estree.js

```diff
@@ -5077,6 +5079,37 @@ function printMemberChain(path, options, print) {
   ]);
 }
 
+/**
+ * @param {import('estree').Node} node
+ * @returns {boolean}
+ */
+function isSimple(node) {
+  if (
+    !node ||
+    node.type === "Literal" ||
+    node.type === "BooleanLiteral" ||
+    node.type === "NullLiteral" ||
+    node.type === "NumericLiteral" ||
+    node.type === "StringLiteral" ||
+    node.type === "Identifier"
+  ) {
+    return true;
+  }
+  if (node.type === "TemplateLiteral" && node.expressions.length === 0) {
+    return true;
+  }
+  if (node.type === "ArrayExpression") {
+    return node.elements.every(isSimple);
+  }
+  if (node.type === "CallExpression") {
+    return isSimple(node.callee) && node.arguments.every(isSimple);
```

Comment:
The recursion opens possibility for not really simple nested calls to pass the check. A depth limit is needed.